peps/pep-0777.rst

```diff
@@ -76,22 +76,47 @@ However, resolvers do not currently exclude wheels with an incompatible wheel
 version. There is also currently no way for a resolver to check a wheel's
 version without downloading the wheel directly. To make wheel version filtering
 easy for resolvers, the wheel version **MUST** be included in the relevant
-metadata file (currently METADATA). This will allow resolvers to efficiently
-check the wheel version using the :pep:`658` metadata API without needing to
-download and inspect the ``.dist-info/WHEEL`` file.
-
-To accomplish this, a new core metadata field is introduced called
-``Wheel-Version``. While this field is optional for metadata included in a
-wheel of major version 1, it is a mandatory field for metadata in wheels of major
-version 2 or higher. This enforces that future revisions of the wheel
+metadata file (currently ``METADATA``). This will allow resolvers to
+efficiently check the wheel version using the :pep:`658` metadata API without
+needing to download and inspect the ``.dist-info/WHEEL`` file.
+
+To accomplish this, a new field, ``Wheel-Version``, will be added to the
+`Core Metadata Specification <https://packaging.python.org/specifications/core-metadata/>`_.
+This field is single use, and must contain the exact same version specified as
+the ``Wheel-Version`` entry in the ``WHEEL`` file, or any future replacement
+file defining metadata about the wheel file. If ``Wheel-Version`` is absent
+from the metadata file, then tools **MUST** infer the wheel file major
+version as 1.
+
+``Wheel-Version`` **MUST NOT** be included in source distribution metadata
+(``PKG-INFO``) files. If a tool encounters ``Wheel-Version`` inside of a source
+distribution metadata file, it **SHOULD** raise an error.
+
+``Wheel-Version`` **MAY** be included in the metadata file for wheels of
+version 1, but for wheels of version 2 or higher, the metadata file **MUST**
+include ``Wheel-Version``. This enforces that future revisions of the wheel
 specification can rely on resolvers skipping incompatible wheels by checking
-the ``Wheel-Version`` field.
-
-The ``Wheel-Version`` field in the metadata file shall contain the exact same entry as the
-``Wheel-Version`` entry in the ``WHEEL`` file, or any future replacement file
-defining metadata about the wheel file. Installers **MUST** verify that these
-entries match when installing a wheel. If ``Wheel-Version`` is absent from the
-metadata file, then the implied major version of the wheel is 1.
+the ``Wheel-Version`` field. Build backends are encouraged to include
+``Wheel-Version`` in all wheels that they generate, regardless of version.
+
+When installing a wheel, installers **MUST NOT** remove the ``Wheel-Version``
+in the installed metadata file. If this was allowed, the ``RECORD`` file would
```

Comment:
I don't follow the "if this was allowed" reasoning and tbh don't remember this issue from the discussion thread.  Could you elaborate or add a link to the deeper discussion about why this must be so?